src/exporters/prometheus.rs

```diff
@@ -1,192 +1,136 @@
 //! # PrometheusExporter
 //!
-//! `PrometheusExporter` implementation, expose metrics to
-//! a [Prometheus](https://prometheus.io/) server.
-use super::utils::get_hostname;
+//! The Prometheus Exporter expose metrics to a [Prometheus](https://prometheus.io/) server.
+//! This is achieved by exposing an HTTP endpoint, which the Prometheus will
+//! [scrape](https://prometheus.io/docs/prometheus/latest/getting_started).
+
+use super::utils;
 use crate::current_system_time_since_epoch;
 use crate::exporters::{Exporter, MetricGenerator, MetricValueType};
 use crate::sensors::{Sensor, Topology};
 use chrono::Utc;
-use clap::{Arg, ArgMatches};
 use hyper::service::{make_service_fn, service_fn};
 use hyper::{Body, Request, Response, Server};
 use std::convert::Infallible;
-use std::fmt::Write as _;
 use std::{
     collections::HashMap,
-    net::{IpAddr, SocketAddr},
+    fmt::Write,
+    net::{IpAddr, SocketAddr, Ipv4Addr},
     sync::{Arc, Mutex},
     time::Duration,
 };
 
 /// Default ipv4/ipv6 address to expose the service is any
-const DEFAULT_IP_ADDRESS: &str = "::";
+const DEFAULT_IP_ADDRESS: IpAddr = IpAddr::V4(Ipv4Addr::new(127, 0, 0, 1));
```

Comment:
not sure about this, :: actually works for both ipv4 and ipv6 on most os now iirc

Reply:
good catch, `::` is actually full zero! localhost would be `::1`.
Does using `::` work with IPv6 disabled? If not, I could use the IPv4 version `0.0.0.0` instead.

Reply:
I've edited the commit to use `0.0.0.0` and fix clippy warnings